### PR TITLE
Enable half communication for distributed

### PR DESCRIPTION
**Pull request by csarofeen**
Add's cuda half support in reference to https://github.com/pytorch/pytorch/issues/2699

### CODE REVIEW

File: torch/csrc/distributed/Module.cpp [hunk 1]
```diff
@@ -214,14 +214,15 @@ THDTensorDescriptor THDPModule_makeDescriptor(PyObject *obj)
...
+  REGISTER_THC_DESCRIPTOR(HalfTensor);
   REGISTER_THC_DESCRIPTOR(LongTensor);
   REGISTER_THC_DESCRIPTOR(IntTensor);
   REGISTER_THC_DESCRIPTOR(ShortTensor);
   REGISTER_THC_DESCRIPTOR(CharTensor);
   REGISTER_THC_DESCRIPTOR(ByteTensor);
 #undef REGISTER_THC_DESCRIPTOR
 #endif
-  throw std::runtime_error(std::string("don't know how to create a THDTensorDesciptor for "
+  throw std::runtime_error(std::string("don't know how to create a THDTensorDesciptor for HALF "
```
Comment (apaszke) [inline torch/csrc/distributed/Module.cpp]: This doesn’t look correct

**Review by apaszke — COMMENTED** [review]
Please just fix the error message. Apart from this LGTM

**Comment by csarofeen** [comment]
Sorry about that. Thanks Adam.

**Review by apaszke — APPROVED** [review]

**Comment by csarofeen** [comment]
Added half for gloo as well.

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
All looks good, but the linter has failed

**Pull request merged**